### ISSUE TITLE
[pkg/winperfcounters] Possible Invalid Data with Wildcard Scraping

### ISSUE THREAD

**Issue by alxbl** [open-telemetry/opentelemetry-collector-contrib#32798]
### Component(s)

pkg/winperfcounters

## Description
> **CONTEXT**: I am opening this issue to accompany a PR which adds support for resetting `pkg/winperfcounters` `watcher` objects in order to "fix" corrupted PDH queries.
 
We have a production deployment where we use `windowsperfcounters` to monitor several host metrics, including `\Process\% Processor Time` for a subset of processes that we care about. One of our support engineers noticed that on a subset of systems, the counter value was way out of line (e.g. perfmon displaying 0.325 while scraped value is 430327.93).

I wrote a testjig that uses `pkg/winperfcounters` to watch some counters and I was able to reproduce the behavior (See logs)

```go
package main

import (
	"flag"
	"log"
	"time"

	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/winperfcounters"
)

func main() {
	log.Print("Starting...")
	category := flag.String("category", "Process", "counter category to query")
	counter := flag.String("counter", "% Processor Time", "counter name to query")
	instance := flag.String("instance", "powershell*", "instance pattern to query (supports wildcards)")
	interval := flag.String("interval", "10s", "scrape interval as a Duration (30s, 1m, etc.)")
	flag.Parse()

	watcher, err := winperfcounters.NewWatcher(*category, *instance, *counter)
	if err != nil {
		log.Fatalf("failed to initialize watcher.\n  %s", err)
	}

	log.Printf("Scraping '%s\\%s' instance=%s every %s...", *category, *counter, *instance, *interval)
	duration, err := time.ParseDuration((*interval))
	if err != nil {
		panic("invalid duration in --interval")
	}

	ticker := time.NewTicker(duration)
	done := make(chan bool)
	for {
		select {
		case <-done:
			return
		case <-ticker.C:
			data, err := watcher.ScrapeData()
			if err != nil {
				log.Printf("Failed scrape: %s", err)
			}

			log.Print("-- Scrape Results --")
			for i, val := range data {
				log.Printf("  %02d %s: %f", i, val.InstanceName, val.Value)
			}
		}
	}
}
```

## Steps to Reproduce
Unclear how the issue occurs, but the following are known minimum requirements 
- Two or more counters which systematically always return `negative denominator` errors.
- Wildcard counter on a multi-instance counter which includes the counters mentioned above and some "healthy" counters
- Working load on the machine (Unsure, but we were only able to reproduce the issue on machines with production workloads)

## Expected Result
One of:
- No data at all for the invalid counters
- Same data as reported in perfmon

## Actual Result
- A single completely wrong value for all counters that are in the `negative denominator` state.

### Collector version

0.98.0-dev

## Environment
OS: Windows Server 2012
Compiler: go version go1.22.2 windows/amd64

### OpenTelemetry Collector configuration

```yaml
N/A
```

### Log output

```shell
[redacted/edited for confidentiality]

	PS C:\Users\[redacted]\Desktop> .\winperfchk.exe --instance='G[redacted]*'
	2024/04/25 14:28:16 Starting...
	2024/04/25 14:28:16 Scraping 'Process\% Processor Time' instance=G[redacted]* every 10s...
// First Scrape
	2024/04/25 14:28:26 GetFormattedCounterArrayDouble - Count=5 BuffSize=0000018c
	2024/04/25 14:28:26   00> Status=00000000
	2024/04/25 14:28:26   01> Status=00000000
	2024/04/25 14:28:26   02> Status=00000000
	2024/04/25 14:28:26   03> Status=00000000
	2024/04/25 14:28:26   04> Status=00000000
	2024/04/25 14:28:26 Scrape Result
	2024/04/25 14:28:26   00 G[InvalidCounterInstance]: 1.249613
	2024/04/25 14:28:26   01 G[InvalidCounterInstance]_1: 6.248066
	2024/04/25 14:28:26   02 G[InvalidCounterInstance]_2: 2.030622
	2024/04/25 14:28:26   03 G[ValidCounterA]_3: 0.781008
	2024/04/25 14:28:26   04 G[ValidCounterB]_4: 0.156202
// Second Scrape
	2024/04/25 14:28:36 GetFormattedCounterArrayDouble - Count=5 BuffSize=0000018c
	2024/04/25 14:28:36   00> Status=c0000bba  // <- INVALID_DATA for G[InvalidCounterInstance]
	2024/04/25 14:28:36   01> Status=00000000  // <- SUCCESS for G[InvalidCounterInstance]_1
	2024/04/25 14:28:36   02> Status=c0000bba  // <- INVALID_DATA for G[InvalidCounterInstance]_2
	2024/04/25 14:28:36   03> Status=00000000
	2024/04/25 14:28:36   04> Status=00000000
	2024/04/25 14:28:36 Scrape Result
	2024/04/25 14:28:36   00 G[InvalidCounterInstance]_1: 468838.396205
	2024/04/25 14:28:36   01 G[ValidCounterA]_3: 2.030732
	2024/04/25 14:28:36   02 G[ValidCounterB]_4: 0.000000
```

### Additional context

Here is a screenshot of the testjig when querying only "corrupted" counters. The very first scrape always seems to work and return proper values. Subsequent scrapes always return the `negative value` error:
![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/860411/c27ede71-a704-4c48-afda-58e91402a52f)

I instrumented some of the `watcher` code to dump the PDH API return status.

I also made two interesting observations while troubleshooting:

1. This never happens on the very first `GetFormattedCounterArrayDouble()` call.
2. This never happens when monitoring a single counter with a specific instance (including the index)

Telegraf appears to not be affected because they do wildcard expansion (and thus fall under scenario 2). The downside of wildcard expansion is that new instances will not be picked up on subsequent scrapes because of explicit instance indices.

I have a PR coming up which adds an API to Re-initialize a PDH query on each scrape, based on observation 1. This seems to fix the issue.

**Comment by crobert-1** [comment]
_Filed by code owner, removing `needs triage`._

**Comment by alxbl** [comment]
Fixed by #32799
